sdks/java/core/src/test/java/org/apache/beam/sdk/testing/BigqueryMatcherTest.java

```diff
@@ -66,62 +63,55 @@
-    when(mockQuery.execute()).thenReturn(response);
+    when(mockQuery.execute()).thenReturn(createResponseContainsTestData());
 
     assertThat(mockResult, matcher);
     verify(matcher).newBigqueryClient(eq(appName));
     verify(mockJobs).query(eq(projectId), eq(new QueryRequest().setQuery(query)));
   }
 
   @Test
-  public void testBigqueryMatcherFailsForServiecFails() throws IOException {
-    BigqueryMatcher matcher = spy(new BigqueryMatcher(
-        appName, projectId, query, "some-checksum"));
+  public void testBigqueryMatcherFailsForChecksumMismatch() throws IOException {
+    BigqueryMatcher matcher = spy(
+        new BigqueryMatcher(appName, projectId, query, "incorrect-checksum"));
     doReturn(mockBigqueryClient).when(matcher).newBigqueryClient(anyString());
-    when(mockBigqueryClient.jobs()).thenReturn(mockJobs);
-    when(mockJobs.query(anyString(), any(QueryRequest.class))).thenReturn(mockQuery);
-    when(mockQuery.execute()).thenThrow(new IOException());
+    when(mockQuery.execute()).thenReturn(createResponseContainsTestData());
 
-    thrown.expect(RuntimeException.class);
-    thrown.expectMessage("Failed to retrieve BigQuery data.");
-    assertThat(mockResult, matcher);
-    verify(matcher).newBigqueryClient(eq(appName));
-    verify(mockJobs).query(eq(projectId), eq(new QueryRequest().setQuery(query)));
+    try {
+      assertThat(mockResult, matcher);
+    } catch (AssertionError expected) {
+      assertThat(expected.getMessage(), containsString("Total number of rows are: 1"));
+      assertThat(expected.getMessage(), containsString("abc"));
+      verify(matcher).newBigqueryClient(eq(appName));
+      verify(mockJobs).query(eq(projectId), eq(new QueryRequest().setQuery(query)));
+    }
   }
 
   @Test
-  public void testBigqueryMatcherFailsForInvalidQueryResponse() throws IOException {
-    BigqueryMatcher matcher = spy(new BigqueryMatcher(
-        appName, projectId, query, "some-checksum"));
+  public void testBigqueryMatcherFailsForServiecFails() throws IOException {
+    BigqueryMatcher matcher = spy(
+        new BigqueryMatcher(appName, projectId, query, "some-checksum"));
     doReturn(mockBigqueryClient).when(matcher).newBigqueryClient(anyString());
-    when(mockBigqueryClient.jobs()).thenReturn(mockJobs);
-    when(mockJobs.query(anyString(), any(QueryRequest.class))).thenReturn(mockQuery);
-    when(mockQuery.execute()).thenReturn(null);
+    when(mockQuery.execute()).thenThrow(new IOException());
 
     try {
       assertThat(mockResult, matcher);
     } catch (AssertionError expected) {
-      assertThat(expected.getMessage(), containsString("was (null)"));
+      assertThat(expected.getMessage(), containsString("BigQuery response is null"));
       verify(matcher).newBigqueryClient(eq(appName));
-      verify(mockJobs).query(eq(projectId), eq(new QueryRequest().setQuery(query)));
+      verify(mockJobs, times(4)).query(eq(projectId), eq(new QueryRequest().setQuery(query)));
```

Comment:
This test will take multiple seconds to execute; one of the benefits of using Sleeper is that you can mock/fake it, verify the same type of things (was the sleep method called n times, etc) but it doesn't actually sleep.